sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_authentication_async.py

```diff
@@ -25,14 +25,23 @@
 # --------------------------------------------------------------------------
 from typing import TYPE_CHECKING
 
-from azure.core.pipeline.policies import AsyncBearerTokenCredentialPolicy
+from azure.core.pipeline.policies import AsyncBearerTokenCredentialPolicy, AsyncHTTPPolicy
 
-from ._authentication import _parse_claims_challenge
+from ._authentication import _parse_claims_challenge, _AuxiliaryAuthenticationPolicyBase
 
 if TYPE_CHECKING:
     from azure.core.pipeline import PipelineRequest, PipelineResponse
 
 
+async def await_result(func, *args, **kwargs):
+    """If func returns an awaitable, await it."""
+    result = func(*args, **kwargs)
+    if hasattr(result, "__await__"):
+        # type ignore on await: https://github.com/python/mypy/issues/7587
+        return await result  # type: ignore
+    return result
+
+
```

Comment:
Hi @xiangyan99 Is `AuxiliaryAuthenticationPolicy` only for mgmt? I think it is better to put the code in `azure-core` so that we don't the duplicated code here.

Reply:
I am not aware of any data plane SDK that needs the "x-ms-authorization-auxiliary" header.

Do you know any?

I am thinking we can start with making it in the mgmt-core and if, in the future, we need to move it into core, we can update it in a non-breaking way.

Reply:
@xiangyan99  approve for the PR.